lib/Service/ImportService.php

```diff
@@ -234,26 +239,40 @@ private function createRow(Row $row): void {
 	}
 
 	/**
-	 * @param Row $row
+	 * @param Row $firstRow
+	 * @param Row $secondRow
 	 * @throws InternalError
 	 * @throws NotFoundError
 	 * @throws PermissionError
 	 * @throws DoesNotExistException
 	 * @throws MultipleObjectsReturnedException
 	 */
-	private function getColumns(Row $row): void {
-		$cellIterator = $row->getCellIterator();
+	private function getColumns(Row $firstRow, Row $secondRow): void {
+		$cellIterator = $firstRow->getCellIterator();
+		$secondRowCellIterator = $secondRow->getCellIterator();
 		$titles = [];
+		$dataTypes = [];
 		foreach ($cellIterator as $cell) {
 			if ($cell && $cell->getValue() !== null && $cell->getValue() !== '') {
 				$titles[] = $cell->getValue();
+				$originDataType = $secondRowCellIterator->current()->getDataType();
+
+				// Convert data type to our data type
+				if (Date::isDateTime($secondRowCellIterator->current()) || $originDataType === 'd') {
```

Comment:
```suggestion
				if ($originDataType === DataType::TYPE_ISO_DATE || Date::isDateTime($secondRowCellIterator->current())) {
```

Let's also use the contant on the number check below...